apps/remote_control/lib/lexical/remote_control/code_intelligence/rename.ex

```diff
@@ -0,0 +1,164 @@
+defmodule Lexical.RemoteControl.CodeIntelligence.Rename do
+  alias Lexical.Ast
+  alias Lexical.Document
+  alias Lexical.Document.Edit
+  alias Lexical.Document.Line
+  alias Lexical.Document.Position
+  alias Lexical.RemoteControl.CodeIntelligence.Entity
+  alias Lexical.RemoteControl.Search.Store
+  require Logger
+
+  import Line
+
+  @spec rename(Document.t(), Position.t(), String.t()) ::
+          {:ok, %{Lexical.uri() => [Edit.t()]}} | {:error, term()}
+  def rename(%Document{} = document, %Position{} = position, new_name) do
+    with {:ok, entity, range} <- resolve(document, position),
+         {:ok, results} <- search(document, position, entity, range) do
+      to_uri_with_changes(results, new_name)
+    end
+  end
+
+  defp resolve(document, position) do
+    case Entity.resolve(document, position) do
+      {:ok, {:module, module}, range} -> {:ok, module, range}
+      {:error, error} -> {:error, error}
+    end
+  end
+
+  defp search(document, position, entity, range) do
+    cursor_entity_string = cursor_entity_string(range)
+
+    if at_the_middle_of_module?(document, position, range) do
+      search_descendants(entity, cursor_entity_string)
+    else
+      search_exact(entity, cursor_entity_string)
+    end
+  end
+
+  defp at_the_middle_of_module?(document, position, range) do
+    range_text = range_text(range)
+
+    case Ast.surround_context(document, position) do
+      {:ok, %{context: {:alias, alias}}} ->
+        String.length(range_text) < length(alias)
+
+      _ ->
+        false
+    end
+  end
+
+  defp cursor_entity_string(range) do
+    # Parent.|Module -> Module
+    range
+    |> range_text()
+    |> String.split(".")
+    |> List.last()
+  end
+
+  defp search_descendants(entity, cursor_entity_string) do
+    entity_string = inspect(entity)
+
+    case Store.fuzzy(entity_string, subject: [:definition, :reference]) do
```

Comment:
I think you should be using `exact` rather than fuzzy here. Fuzzy will return a lot more results.

Reply:
But I'm looking for `descendants`, is there a better way than fuzzy?